src/robust_provider/mod.rs

```diff
@@ -1,3 +1,54 @@
+//! Robust, retrying wrapper around Alloy providers.
+//!
+//! This module exposes [`RobustProvider`], a small wrapper around Alloy's
+//! `RootProvider` that adds:
+//! * bounded per-call timeouts,
+//! * exponential backoff retries,
+//! * transparent failover between a primary and one or more fallback providers,
+//! * more robust WebSocket block subscriptions with automatic reconnection.
+//!
+//! Use [`RobustProviderBuilder`] to construct a provider with sensible defaults
+//! and optional fallbacks, or use the [`IntoRobustProvider`] and [`IntoProvider`]
+//! traits to upgrade common provider types.
+//!
+//! # How it works
+//!
+//! All RPC calls performed through [`RobustProvider`] are wrapped in a total
+//! timeout (`call_timeout`) and retried with exponential backoff up to
+//! `max_retries`. If the primary provider keeps failing, the call is retried
+//! against the configured fallback providers in order. For subscriptions,
```

Comment:
```suggestion
//! against the configured fallback providers in the order they were added. For subscriptions,
```
or something like that